src/compiler/parser.ts

```diff
@@ -6259,7 +6259,7 @@ namespace ts {
                 }
 
                 function parseJSDocIdentifier(): Identifier {
-                    if (token !== SyntaxKind.Identifier) {
+                    if (token !== SyntaxKind.Identifier && !isTSOnlyKeyword(token)) {

```

Comment:
what is wrong with `tokenIsIdentifierOrKeyword(token)`? do JS keywords appear in the grammar in a valid position?